cmd/grafana-backups/main_test.go

```diff
@@ -0,0 +1,130 @@
+package main
+
+import (
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+)
+
+func TestWriteDashboardFile(t *testing.T) {
+	expected := "Successful Response"
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path == "/api/dashboards/uid/fakeUID" {
+
+			_, err := io.WriteString(w, expected)
+
+			if err != nil {
+				t.Fatal(err)
+			}
+		} else {
+			_, err := io.WriteString(w, "Bad response for providing an incorrect path")
+
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+	}))
+	defer ts.Close()
+
+	dir, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+
+	err = writeDashboardFile(dir, "fakeUID", ts.URL, "test")
+	if err != nil {
+		t.Errorf("writeDashboardFile failed")
+	}
+	content, err := ioutil.ReadFile(dir + "/fakeUID.json")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if string(content) != "Successful Response" {
+		t.Errorf("Expected %q. Got %q", expected, string(content))
+	}
+}
+
+func TestWriteDashboardFileError(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(500)
+	}))
+	defer ts.Close()
+
+	dir, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+
+	err = writeDashboardFile(dir, "FakeUID", ts.URL, "test")
+	if err == nil {
+		t.Error("Expected error, got none")
+	}
+	if !strings.Contains(err.Error(), "Invalid response code") {
+		t.Errorf("Expected 'Invalid response code', got %q", err)
+	}
+	_, err = os.Stat(dir + "/FakeUID")
+	if err == nil {
+		t.Error("Expected error, got none")
+	}
+	if !os.IsNotExist(err) {
+		t.Errorf("Expected `no such file or directory`  got %q", err)
```

Comment:
nit: extra space in format string
```suggestion
		t.Errorf("Expected `no such file or directory` got %q", err)
```